src/lib/components/LighthouseScoresAudits/index.js

```diff
@@ -0,0 +1,244 @@
+import {demoLhr} from "./demo";
+
+const NUM_AUDITS_TO_SHOW = 10;
+
+/**
+ * Renders a table for the passed Lighthouse report.
+ *
+ * @param {!LighthouseLastRunDetail} lhr The lighthouse report to render a table for.
+ * @param {!CombinedAuditAndAuditRef} category
+ * @return {!TemplateResult}
+ */
+function createRowForAuditCategory(lhr, category) {
+  let relevantGuides = ["Guide coming soon"];
+  const audit = lhr.audits[category.ref.id];
+
+  const auditGuideMapping = getAuditGuideMapping();
+  if (auditGuideMapping) {
+    const guidesForAudit = auditGuideMapping[category.ref.id];
+    if (guidesForAudit) {
+      relevantGuides = guidesForAudit.map((guide) => {
+        return html`
+          <a
+            href="${guide.url}"
+            class="lh-audit-list-row__link internal gc-analytics-event"
+            data-category="web.dev"
+            data-label="audit todos, internal link"
+            data-action="click"
+            >${guide.title}</a
+          >
+        `;
+      });
+    } else {
+      const docLink = getAuditReferenceDocLink(audit.description);
+      if (docLink) {
+        relevantGuides = html`
+          <a
+            href="${docLink}"
+            class="lh-audit-list-row__link external gc-analytics-event"
+            target="_blank"
+            data-category="web.dev"
+            data-label="audit todos, external link"
+            data-action="click"
+            >${audit.title}</a
+          >
+        `;
+      }
+    }
+  }
+
+  return html`
+    <div class="lh-audit-list-row" role="row">
+      <span
+        class="lh-audit-list-row__impact lh-audit-list-row__item
+                lh-audit-list-row__impact--${category.impact.toLowerCase()}"
+        role="cell"
+        >${category.impact}</span
+      >
+      <span
+        class="lh-audit-list-row__item lh-audit-list-row__category"
+        role="cell"
+        >${category.ref.cat}</span
+      >
+      <span
+        class="lh-audit-list-row__item lh-audit-list-row__recommendation"
+        role="cell"
+        >${audit.title}</span
+      >
+      <span class="lh-audit-list-row__item lh-audit-list-row__guide" role="cell"
+        >${relevantGuides}</span
+      >
+    </div>
+  `;
+}
+
+/* eslint-disable require-jsdoc */
+class LighthouseScoresAudits extends BaseElement {
+  static get properties() {
+    return {
+      timesExpanded: {type: Number},
+      inverted: {type: Boolean},
+      filteringOn: {type: String},
+      lhr: {type: Object},
+      demo: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.timesExpanded = 0;
+    this.inverted = false;
+
+    this.setAttribute("role", "table");
+    this.setAttribute("aria-label", "Lighthouse audit results");
+  }
+
+  firstUpdated() {
+    // FIXME FIXME FIXME: remove this demo before submit
+    if (this.demo) {
+      this.lhr = JSON.parse(demoLhr)[0].lhr;
+    }
+  }
+
+  update(changedProperties) {
+    if (changedProperties.has("lhr")) {
```

Comment:
This doesn't work because the property change handler isn't called with bound `this`. It has to be a pure function.

I've rewritten it to provide our own getter and setter. WDYT?

Reply:
oh ok. I'm cool with getter/setter. thank you!